Tests/SwiftDocCTests/Model/DocumentationNodeTests.swift

```diff
@@ -0,0 +1,39 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import Markdown
+@testable import SwiftDocC
+import XCTest
+
+class DocumentationNodeTests: XCTestCase {
+    func testH4AndUpAnchorSections() throws {
```

Comment:
Should we also check the contents of the anchor sections to make sure they match the title in source?

It would also be good to have an integration-style test that verifies that the render JSON contains the right anchors as well.

Reply:
Got it. I'll update the test case later and then request a new review.

Reply:
> Should we also check the contents of the anchor sections to make sure they match the title in source?

Done.

> It would also be good to have an integration-style test that verifies that the render JSON contains the right anchors as well.

Looks that it would always be emitted to the render JSON even before this PR. Do we  still need to check it?
https://github.com/apple/swift-docc/blob/61100a442ff1021405e19de8b4b225e576b19cdd/Sources/SwiftDocC/Model/Rendering/RenderContentCompiler.swift#L48-L50

But I do think we should check if the following article has no problems when rendering

```swift
<doc:HeadingTest#Heading2>
<doc:HeadingTest#Heading3>
<doc:HeadingTest#Heading4>
<doc:HeadingTest#Heading5>
<doc:HeadingTest#Heading6>
```

I have some problems to do the integration test for this. Could you provide some suggestion on this?

```swift
func testH4AndUpAnchorSections() throws {
    let articleSource = """
    # HeadingAnchorTest

    ## Heading2

    ### Heading3
    
    #### Heading4
    
    ##### Heading5

    ###### Heading6
    """
    let article = Article(markup: Document(parsing: articleSource, options: []), metadata: nil, redirects: nil, options: [:])
    let (_, bundle, context) = try testBundleAndContext(copying: "TestBundle") { root in
        try articleSource.write(to: root.appendingPathComponent("HeadingAnchorTest.md"), atomically: true, encoding: .utf8)
        try """
        # HeadingAnchorTest2

        <doc:HeadingTest#Heading2>
        <doc:HeadingTest#Heading3>
        <doc:HeadingTest#Heading4>
        <doc:HeadingTest#Heading5>
        <doc:HeadingTest#Heading6>
        """.write(to: root.appendingPathComponent("HeadingAnchorTest2.md"), atomically: true, encoding: .utf8)
    }
    
    // FIXME: Could not get the node from documentationCache
//        let node = try context.entity(with: ResolvedTopicReference(bundleIdentifier: "org.swift.docc.example", path: "/documentation/HeadingAnchorTest", sourceLanguage: .swift))
    let node = try DocumentationNode(reference: ResolvedTopicReference(bundleIdentifier: "org.swift.docc.example", path: "/documentation/HeadingAnchorTest", sourceLanguage: .swift), article: article)
    XCTAssertEqual(node.anchorSections.count, 5)
    for (index, anchorSection) in node.anchorSections.enumerated() {
        let expectedTitle = "Heading\(index + 2)"
        XCTAssertEqual(anchorSection.title, expectedTitle)
        XCTAssertEqual(anchorSection.reference, node.reference.withFragment(expectedTitle))
    }
    let unresolvedTopicProblems = context.problems.filter { $0.diagnostic.identifier == "org.swift.docc.unresolvedTopicReference" }
    // FIXME: This it not going to work since there is other unresolvedTopicReference in TestBundle. How can we get a bundle and context only have this 2 articles?
    XCTAssertEqual(unresolvedTopicProblems.count, 0)
}
```

Reply:
Sorry for the delay on this. You're right, since this PR doesn't affect how the headings are propagated through the render JSON, there's no need for an additional text.